core/modules/commands/server.js

```diff
@@ -91,57 +91,59 @@ SimpleServer.prototype.checkCredentials = function(request,incomingUsername,inco
 	}
 };
 
-SimpleServer.prototype.listen = function(port,host) {
-	var self = this;
-	http.createServer(function(request,response) {
-		// Compose the state object
-		var state = {};
-		state.wiki = self.wiki;
-		state.server = self;
-		state.urlInfo = url.parse(request.url);
-		// Find the route that matches this path
-		var route = self.findMatchingRoute(request,state);
-		// Check for the username and password if we've got one
-		var username = self.get("username"),
-			password = self.get("password");
-		if(username && password) {
-			// Check they match
-			if(self.checkCredentials(request,username,password) !== "ALLOWED") {
-				var servername = state.wiki.getTiddlerText("$:/SiteTitle") || "TiddlyWiki5";
-				response.writeHead(401,"Authentication required",{
-					"WWW-Authenticate": 'Basic realm="Please provide your username and password to login to ' + servername + '"'
-				});
-				response.end();
-				return;
-			}
-		}
-		// Return a 404 if we didn't find a route
-		if(!route) {
```

Comment:
This will bind the `requestHandler` to `window` (global). is that the intent or did you mean: `this.requestHandler.bind(this,this)`? Also what is the browser support for `bind`? Do we need this instead:

```js
SimpleServer.prototype.listen = function(port,host) {
	var self = this;
	http.createServer(function() { return self.requestHandler(self); }).listen(port,host);
};
```